client/src/offline-settings/mdn-worker.tsx

```diff
@@ -188,19 +85,30 @@ export class MDNWorker {
     }
   }
 
-  async updateAvailable() {
-    await this.getUpdate();
-    return this.status();
-  }
-  status() {
-    return this.updateStatus;
+  async status() {
+    return await getContentStatus();
   }
 
   offlineSettings(): SettingsData {
-    return (
+    const settings =
       JSON.parse(window.localStorage.getItem("MDNSettings") || "null") ??
-      new SettingsData()
-    );
+      new SettingsData();
+
+    if (settings.currentDate && settings.currentVersion) {
+      console.log(
+        `[worker] Migrate LocalContentStatus from LocalStorage to IndexedDB`
+      );
+      patchContentStatus({
+        local: {
+          version: settings.currentVersion,
+          date: settings.currentDate,
+        },
+      });
+      delete settings.currentDate;
+      delete settings.currentVersion;
```

Comment:
should we write to local storage here? It's not guaranteed that `setOfflineSettings` is called, is it?

Reply:
You mean calling setOfflineSettings() directly here after having migrated those values?

Reply:
`setOfflineSettings` calls this so we need to directly write it back of we get a loop.

Reply:
Fixed in d021e36a1.